File: mobile_microapps_architecture_flutter/lib/app_shell/composition_root.dart
import 'package:auth_module/auth_module.dart';
import 'package:core_network/core_network.dart';
import 'package:insurance_module/insurance_module.dart';
import 'package:module_contracts/module_contracts.dart';
import 'package:payments_module/payments_module.dart';
import 'package:profile_module/profile_module.dart';

import 'app_session_controller.dart';

final class DemoCompositionRoot {
  DemoCompositionRoot()
      : session = AppSessionController(),
        network = const FakeEnterpriseGateway() {
    registry = ModuleRegistry([
      AuthModule(session: session, network: network),
      PaymentsModule(session: session, network: network),
      InsuranceModule(session: session, network: network),
      ProfileModule(session: session),
    ]);
  }

  final AppSessionController session;
  final NetworkClient network;
  late final ModuleRegistry registry;
}

File: MobileMicroappsArchitecture/src/app_shell/CompositionRoot.ts
import {createAuthModule} from '../packages/auth_module';
import {FakeEnterpriseGateway} from '../packages/core_network';
import {createInsuranceModule} from '../packages/insurance_module';
import {ModuleRegistry} from '../packages/module_contracts';
import {createPaymentsModule} from '../packages/payments_module';
import {createProfileModule} from '../packages/profile_module';

import {AppSessionController} from './AppSessionController';

export class DemoCompositionRoot {
  readonly session = new AppSessionController();
  readonly network = new FakeEnterpriseGateway();
  readonly registry = new ModuleRegistry([
    createAuthModule({session: this.session, network: this.network}),
    createPaymentsModule({session: this.session, network: this.network}),
    createInsuranceModule({session: this.session, network: this.network}),
    createProfileModule({session: this.session}),
  ]);
}
